apis/src/components/organisms/quickplay.rs

```diff
@@ -0,0 +1,83 @@
+use core::panic;
+use leptos::*;
+
+use leptos_router::use_navigate;
+
+use crate::{
+    common::ChallengeAction,
+    providers::{ApiRequests, AuthContext},
+};
+use hive_lib::{ColorChoice, GameType};
+use shared_types::ChallengeVisibility;
+use shared_types::{ChallengeDetails, TimeMode};
+
+#[component]
+fn GridButton(text: String) -> impl IntoView {
```

Comment:
Knowing leex I suspect he'll want to use an enum here instead of text: String, or something like default_speed: Type.